Fi 354 bulk data base sequence

yunwwang opened the pull request:
This pull request contains changes to move core functions from BulkDataPateintExportSequence to BulkDataExportSequence.

**Submitter:**
- [x] This pull request describes why these changes were made
- [x] Internal ticket for this PR: FI-354 Add base class for bulk data export
- [x] Internal ticket links to this PR
https://oncprojectracking.healthit.gov/support/browse/FI-354
- [x] Internal ticket is properly labeled (Community/Program)
- [x] Internal ticket has a justification for its Community/Program label
- [x] Code diff has been reviewed for extraneous/missing code
- [x] Tests are included and test edge cases
- [x] Tests/code quality metrics have been run locally and pass


**Reviewer 1:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

**Reviewer 2:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

Jammjammjamm commented on lib/app/modules/bulk_data/bulk_data_patient_export_sequence.rb:
```diff
@@ -1,303 +1,25 @@
+      title 'Patient Compartment Export Tests'
 
-      description 'Verify that Patient resources on the Bulk Data server follow the Bulk Data Access Implementation Guide'
+      description 'Verify that patient compartment export on the Bulk Data server follow the Bulk Data Access Implementation Guide'
```
```suggestion
      description 'Verify that patient compartment export on the Bulk Data server follows the Bulk Data Access Implementation Guide'
```

Jammjammjamm commented on lib/app/modules/bulk_data/bulk_data_export_sequence.rb:
```diff
@@ -0,0 +1,305 @@
+
+      private
+
+      def export_kick_off(endpoint = nil,
```
It's confusing to me that this default value is `nil`, but `endpoint` is set to `''`. If this argument is always being passed in, i think we should get rid of the default value.

yunwwang commented:
> Apart from adding the `endpoint` method, the code is basically the same, right?
Yes.
1) All tests and functions are moved from BulkDataPatientSequence to BulkDataSequence.
2) Removed endpoint parameters from some functions' signature
2) Added endpoint function to be overwritten by child sequence.
